tree/30_generic_methods/_file_check_special.cf

```diff
@@ -35,10 +35,11 @@ bundle agent file_check_special(file_name, file_type)
       "file_has_type"       string => filestat("${file_name}", "type");
 
       "old_class_prefix"    string => canonify("file_check_${file_type}_${file_name}");
-      "promisers"            slist => { @{this.callers_promisers}, cf_null }, policy => "ifdefined";
-      "class_prefix"        string => canonify(join("_", "promisers"));
-      "args"                slist => { "${file_name}", "${file_type}" };
 
+      "args"                 slist => { "${file_name}", "${file_type}" };
+      "report_param"        string => join("_", args);
```

Comment:
Couldn't we have a problem if the sum of variables length is more than max var size? Shouldn't we stay compatible with old_class prefix everywhere?

Reply:
That's an interesting remark.

I tried several generic method with large string, and got failure elsewhere (when running command for instance), but the strings related to classes seem to be correctly truncated when concatenated like that (i only got a crash when I inputed a string larger than 16384 char in the vars: "bla" string => "here" itseld

Reply:
i can't see how we can detect that class_prefix has an issue, and if so, fallback to old_class_prefix

Reply:
it seems to work well indeed, there is no hard limit on var size, but only in the parser.